src/docker.rs

```diff
@@ -351,10 +351,168 @@ pub fn run(
         }
     }
 
+    let mut image = image(config, target)?;
+
+    // TODO: if -p, use that
+    let package = metadata.packages.get(0).unwrap();
+
+    if let Some(path) = config.dockerfile(target)? {
+        let context = config.dockerfile_context(target)?;
+        let name = config.image(target)?;
+
+        let mut build = Dockerfile::File {
+            path: &path,
+            context: context.as_deref(),
+            name: name.as_deref(),
+        };
+
+        if config.dockerfile_provide_base(target).unwrap_or_default() {
+            let mut content = format!("FROM {image}\n");
+            content.push_str(&crate::file::read(&path)?);
+            build = Dockerfile::Stdin {
+                content,
+                from_file: true,
+            };
+        }
```

Comment:
here's the thing I mentioned here https://github.com/cross-rs/cross/pull/678/files#r892810082